### PR TITLE
Comments: Use the comment counts endpoint for pagination.

### LINKED ISSUE #20674: Comments: update pager to use counts endpoint
For the /comments section, let's rely on the new comments count endpoint for values, instead of counting using comments-tree state. For Jetpack sites, let's bump the minimum requirement to Jetpack 5.6

### PR DESCRIPTION
**Pull request by kwight**
This PR uses the new `comment-counts` endpoint as the count source for pagination, rather than the `length` we've been using up to now.

This can be tested on its own as it includes some of #20812 and #20809 .  A rebase will be needed after those land.

**Testing**
* Load `/comments` and check the Networking tab to be sure a request to `comment-counts` was fired.
* Ensure that pagination numbers make sense and match up with the comments being viewed.

Fixes #20674 .

### CODE REVIEW

File: client/my-sites/comments/comment-list/index.jsx [hunk 1]
```diff
@@ -143,8 +149,14 @@ export class CommentList extends Component {
 
 	updateLastUndo = commentId => this.setState( { lastUndo: commentId } );
 
+	getCommentsCount = ( commentCounts, status ) => {
```
Comment (gwwar) [inline client/my-sites/comments/comment-list/index.jsx]: What if we moved this to mapStateToProps, and expose a single count in props for now?

**Comment by gwwar** [comment]
Thanks @kwight I think we might need to land this together with @Copons' PR. I noticed that the counts are correct on my Jetpack site, but I fail to fetch 5 of the last pages 😢

**Comment by kwight** [comment]
Hm, interesting, I'm not having problems with my Jetpack sandbox. Which PR of @copons are you referring to?

**Comment by kwight** [comment]
Closing; these changes have been rolled into #20737 in d7895b20fd67090ed24815c9bd56ffc60118037b.

**Pull request closed without merging**